enumerables.rb

```diff
@@ -0,0 +1,123 @@
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    n_array = []
+    while i < to_a.length
+      x = to_a[i]
+      n_array[i] = yield(x)
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    n_array = []
+    while i < to_a.length
+      x = to_a[i]
+      n_array[i] = yield(x, i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = self
+    n_array = []
+    arr.my_each { |x| n_array << x if yield x }
+    n_array
+  end
+
+  def my_all?(arg = nil)
+    arr = self
+    y = true
+    if block_given?
+      arr.my_each { |x| return false if yield(x) == false }
+      return true
+    elsif !block_given? && arg.nil?
+      return true
+    elsif arg.is_a?(Class)
+      arr.my_each { |x| return false unless x.is_a?(arg) }
+    elsif arg.is_a?(Regexp)
+      arr.my_each { |x| return false unless arg.match(x) }
+    else
+      arr.my_each { |x| return false if x != arg }
+    end
+    y
+  end
```

Comment:
- [ ] When no block or argument is given, `#my_all?` should return false when one of the collection members are false or nil.

```ruby
 [nil, false, true, []].all? === [nil, false, true, []].my_all? => false
```